### PR TITLE
Chat 52  message routing

### PR DESCRIPTION
**Pull request by Troha7**

### CODE REVIEW

File: src/main/java/com/chat/yourway/config/RedisConfig.java [hunk 1]
```diff
@@ -13,6 +13,9 @@
 @Configuration
 public class RedisConfig {
 
+  @Value("${spring.profiles.active}")
+  private String profile;
+
   @Value("${spring.data.redis.host}")
```
Comment (khshanovskyi) [inline src/main/java/com/chat/yourway/config/RedisConfig.java]: if u use more than one @Value in one class or if u use more than once the same @Value property -> move to setting class (@ConfigurationProperties)
https://www.baeldung.com/configuration-properties-in-spring-boot

File: src/main/java/com/chat/yourway/service/ChatMessageServiceImpl.java [hunk 2]
```diff
@@ -48,6 +50,15 @@ public MessageResponseDto sendToContact(MessagePrivateRequestDto message, String
     return messageResponseDto;
   }
 
+  @Override
+  public List<MessageResponseDto> getMessages(Integer topicId) {
```
Comment (khshanovskyi) [inline src/main/java/com/chat/yourway/service/ChatMessageServiceImpl.java:54]: split up with empty row to improve readability

```
    log.trace("Started getMessages for topicId={}", topicId);

    List<MessageResponseDto> messages = messageService.findAllByTopicId(topicId);
    simpMessagingTemplate.convertAndSend(toTopicDestination(topicId), messages);

    log.trace("All messages send to topicId={}", topicId);

    return messages;
```

File: src/test/java/com/chat/yourway/integration/controller/ChatControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    "/dataset/mockdb/topic.xml",
+    "/dataset/mockdb/contact.xml",
+    "/dataset/mockdb/topic_subscriber.xml",
+    "/dataset/mockdb/message.xml"
+}, type = CLEAN_INSERT)
+public class ChatControllerTest {
+
+  @Autowired
+  private ObjectMapper objectMapper;
+  @Autowired
```
Comment (khshanovskyi) [inline src/test/java/com/chat/yourway/integration/controller/ChatControllerTest.java:65]: annotated fields should be split by empty row

File: src/test/java/com/chat/yourway/integration/controller/ChatControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+  private TokenRedisRepository tokenRedisRepository;
+
+  @LocalServerPort
+  private int port;
+  private StompSession session;
+
+  //Test jwt access token for email = vasil@gmail.com, expiration = 19/09/2024
+  private final String JWT_ACCESS_TOKEN = "eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ2YXNpbEBnbWFpbC5jb20iLCJpYXQiOjE2OTc3MTQzNzYsImV4cCI6MTcyNjc0NDc3Nn0.d8cLceS967kh45h2P-cbQix5XAAo4LRYvGUZmo9HFHI";
```
Comment (khshanovskyi) [inline src/test/java/com/chat/yourway/integration/controller/ChatControllerTest.java]: what will we do after 19/09/2024? probably should be some workaround that will help to generate access_token in runtime (also, u need to test some cases with expired token or not valid token)

File: src/test/resources/application.properties [hunk 5]
```diff
@@ -52,4 +52,7 @@ spring.mail.properties.mail.smtp.starttls.enable=true
 spring.mail.properties.mail.smtp.starttls.required=true
 
 # Message values
-message.max.amount.reports=2
\ No newline at end of file
+message.max.amount.reports=2
+
+#Logging:
+logging.level.com.chat.yourway=trace
```
Comment (khshanovskyi) [inline src/test/resources/application.properties:58]: Should prod env support trace logs?

File: src/test/java/com/chat/yourway/integration/controller/ChatControllerTest.java (hunk 4 above)
Comment (Troha7) [reply src/test/java/com/chat/yourway/integration/controller/ChatControllerTest.java]: I can't replace the token during the ws session to test the case with token expiration

**Review by DmytroTeliukov — APPROVED** [review]

**Pull request merged**